src/AspNet.Security.OAuth.Zoho/ZohoAuthenticationHandler.cs

```diff
@@ -0,0 +1,175 @@
+        requestMessage.Version = Backchannel.DefaultRequestVersion;
+
+        using var response = await Backchannel.SendAsync(requestMessage, HttpCompletionOption.ResponseHeadersRead, Context.RequestAborted);
+        if (!response.IsSuccessStatusCode)
+        {
+            await Log.UserProfileErrorAsync(Logger, response, Context.RequestAborted);
+            throw new HttpRequestException("An error occurred while retrieving the user profile.");
+        }
+
+        using var payload = JsonDocument.Parse(await response.Content.ReadAsStringAsync(Context.RequestAborted));
+
+        var principal = new ClaimsPrincipal(identity);
+        var context = new OAuthCreatingTicketContext(principal, properties, Context, Scheme, Options, Backchannel, tokens, payload.RootElement);
+        context.RunClaimActions();
+
+        await Events.CreatingTicket(context);
+        return new AuthenticationTicket(context.Principal!, context.Properties, Scheme.Name);
+    }
+
+    protected override async Task<OAuthTokenResponse> ExchangeCodeAsync(OAuthCodeExchangeContext context)
+    {
+        var nameValueCollection = new Dictionary<string, string?>
+        {
+            ["client_id"] = Options.ClientId,
+            ["client_secret"] = Options.ClientSecret,
+            ["code"] = context.Code,
+            ["redirect_uri"] = context.RedirectUri,
+            ["grant_type"] = "authorization_code"
+        };
+
+        if (context.Properties.Items.TryGetValue(OAuthConstants.CodeVerifierKey, out var codeVerifier))
+        {
+            nameValueCollection.Add(OAuthConstants.CodeVerifierKey, codeVerifier!);
+            context.Properties.Items.Remove(OAuthConstants.CodeVerifierKey);
+        }
+
+        var tokenEndpoint = CreateEndpoint(ZohoAuthenticationDefaults.TokenPath);
+        using var request = new HttpRequestMessage(HttpMethod.Post, tokenEndpoint);
+        request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue(MediaTypeNames.Application.Json));
+        request.Content = new FormUrlEncodedContent(nameValueCollection);
+        request.Version = Backchannel.DefaultRequestVersion;
+
+        using var response = await Backchannel.SendAsync(request, Context.RequestAborted);
+        if (!response.IsSuccessStatusCode)
+        {
+            await Log.ExchangeCodeErrorAsync(Logger, response, Context.RequestAborted);
+            return OAuthTokenResponse.Failed(new Exception("An error occurred while retrieving an access token."));
+        }
+
+        var payload = JsonDocument.Parse(await response.Content.ReadAsStringAsync(Context.RequestAborted));
+
+        return OAuthTokenResponse.Success(payload);
+    }
+
+    /// <summary>
+    /// Creates the endpoint for the Zoho API using the location parameter.
+    /// If the location parameter doesn't match any of the supported locations, the default location (US) is used.
+    /// We don't use the accounts-server parameter due to security reasons.
```

Comment:
Could you elaborate (just here in GitHub is fine, not in the code) why that is?

```suggestion
    /// We don't use the <c>accounts-server</c> parameter for security reasons.
```

Reply:
To prevent client's secrets from being stolen, this summary serves as a reminder of our purpose.